### PR TITLE
feat(ts) refactor to use typescript

### PR DESCRIPTION
**Pull request by darrenjennings**
Converts project to Typescript. When a type was unknown I have used a type alias `type TODO = any` so that it is obvious that it would be preferable to update in the future. I also did not add type definitions for kongponents.js or PieChart.js because of the overwhelming nature of their errors and also because Kongponents should have their own lib level type. This means that you can continue to use `js` files (not coercing users to use `.ts`) but I think that it would be better in the future to resolve all `.js` files and remove `allowJS` in tsconfig.json. This would help with contributors not accidentally checking in untyped js instead of ts files. Also, I did not convert all <script> tags to `lang="ts"`. I will leave this for future PRs. I updated `EntityURLControl.vue` as a single example for now.

Here's what I did:

1. cd /kuma-gui
2. yarn global add @vue/cli
2. vue add typescript
3. answered prompts and converted all js -> ts
4. ran `yarn tsc --noEmit`
5. Added missing types/definitions, fix errors, and add any necessary `@types/<project>` 
6. Removed a ton of dead(unused) code that was failing the type checker. Mostly relics from when the project was originally copied from an internal repo in the Kong org.

I have resolved all type errors, so the code is ready for review.

### CODE REVIEW

File: src/views/Wizard/mixins/updateStorage.ts [hunk 1]
```diff
@@ -1,37 +1,43 @@
...
     this.clearStorage()
   },
   methods: {
-    updateStorage (key, value) {
+    updateStorage (key: string, value: string) {
       // add (or update) item in/to localStorage
-      this.storedVal = value || null
-      this.storedKey = key || null
+      this.storedVal = value || ''
+      this.storedKey = key || ''
```
Comment (darrenjennings) [inline src/views/Wizard/mixins/updateStorage.ts:32]: noteworthy, I am not sure if empty string here is a regression

File: src/views/Wizard/directives/SerializeInput.ts [hunk 2]
```diff
@@ -7,15 +7,15 @@
...
     el.oninput = (e) => {
       const re1 = new RegExp(/[^a-zA-Z0-9 -]/g)
       const re2 = new RegExp(/\s+/g)
-      const re3 = new RegExp(/-+/g)
+      const re3 = new RegExp(/-+/g);
 
       // this will clean up the input value
       // immediately as the user types
-      e.target.value = e.target.value
+      (<HTMLInputElement>e.target).value = (<HTMLInputElement>e.target).value
```
Comment (darrenjennings) [inline src/views/Wizard/directives/SerializeInput.ts:18]: maybe a better way to do this

**Comment by jakubdyszkiewicz** [comment]
Woaaaahhhhh!!!

can't wait for this! 🚀

File: src/router.ts [hunk 3]
```diff
@@ -44,7 +45,7 @@ export default (store) => {
             simpleContent: true,
             onboardingProcess: true
           },
-          component: () => import(/* webpackChunkName: "onboarding-get-started" */ '@/views/Onboarding/GetStarted')
+          component: () => import(/* webpackChunkName: "onboarding-get-started" */ '@/views/Onboarding/GetStarted.vue')
```
Comment (tharun208) [inline src/router.ts:48]: :thumps_up

**Comment by bartsmykla** [comment]
Thank you @darrenjennings for working on this. Could you rebase, as I would love to have it merged?

**Comment by darrenjennings** [comment]
> Thank you @darrenjennings for working on this. Could you rebase, as I would love to have it merged?

@bartsmykla done!

**Review by bartsmykla — APPROVED** [review]
lgtm, @nickolaev @jakubdyszkiewicz wdyt about merging it? :-)

**Review by bartsmykla — APPROVED** [review]
Lgtm, lets merge it :-)

**Pull request merged**